modules/e2ee/E2EEncryption.js

```diff
@@ -263,6 +265,30 @@ export class E2EEncryption {
         case 'e2ee.idKey':
             logger.debug(`Participant ${participant.getId()} updated their id key: ${newValue}`);
             break;
+        case 'e2ee.enabled':
+            this._onParticipantE2EEPropertyChanged(participant, newValue);
+            break;
+        }
+    }
+
+    /**
+     * Handles an update in a participant's e2ee.enabled property changed.
+     * If e2ee disabled, frees the olm session and rotates the key.
+     *
+     * @param {JitsiParticipant} participant - The participant.
+     * @param {*} value - The property's previous value.
+     * @private
+     */
+    _onParticipantE2EEPropertyChanged(participant, value) {
+        if (!value && this._enabled) {
+            const olmData = participant[kOlmData] || {};
```

Comment:
Please move this part to some "clearParticipant" function instead of exposing the inner works.